File: app/models/interest.rb
class Interest < ActiveRecord::Base
  attr_accessible :name

  def self.get_ids_from_omniauth(omniauth)
    
    ids = []
    interest_list = omniauth.extra.raw_info.interests
    
    if interest_list
      interest_list = interest_list.split(',') 
      interest_list.each do |interest|
        ids << self.find_or_create_by_name(interest.strip.downcase).id
      end
    end    
    ids
  end
end

File: app/assets/javascripts/interest.js
function Interest (interest) {
  this.id = null;
  this.name = null;

  var initialize = function (obj, interest) {
    if (interest !== null) {
      obj.id = interest.id;
      obj.name = interest.name;
    }
  };

  initialize(this, interest);
}
